File: 0001-two-sum/0001-two-sum.py
##time: o(n)
##space: o(n)
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashmap = {}
        
        
        for i,n in enumerate(nums):
            difference = target - n
            if difference in hashmap:
                return [hashmap[difference] , i]
            hashmap[n] = i
        return   
    
    
#nums=[2,7,5,11]
#target = 9

#soluiton = Solution()

#result = solution.twoSum(nums,target)
#print(result)
        

File: 0001-two-sum/0001-two-sum.cpp
//time:o(n)
//space:o(n)
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        unordered_map<int,int> m;
        vector<int> result;
        
        for(int i=0;i<nums.size();i++){
            int difference = target - nums[i];
            if(m.find(difference) != m.end()){
                result.push_back(m[difference]);
                result.push_back(i);
            }
            else{
                m.insert({nums[i] , i});
            }
        }
        return result;
        
    }
};